### DISCUSSION TITLE
Compilation failed: "No space left on device" although plenty of space

### DISCUSSION THREAD

**Discussion by aaelberg** [ccache/ccache#1351] [Support]
We have ccache configured to 500GB and no limit on number of files, writing to /tmp (local disk)
From time to time compilation fails with notice:
writing to /tmp/cc<****>.s: No space left on device.
Checking the disk occupancy, it hardly fills 8GB (plenty of free space).
If I copy a file manually, that succeeds.
After I clear the cache (ccache -C), compilation succeeds.

What is the threshold I'm hitting?

Thanks

**Comment by jrosdahl** [comment]
`/tmp/cc<****>.s` is not a file that ccache itself writes, so it's probably a temporary file written by the compiler. I have no idea what happens in your scenario, but a tip is to use `strace -f` to see which process is trying to write to the file.

**Reply by aaelberg** [reply]
@jrosdahl Thanks, already solved the issue. Eventually that was related to diskspace quota and not to ccache
